### PR TITLE
Use armel instead of arm-softfp

### PR DESCRIPTION
**Pull request by hqueue**
Related issue: #8770 

Rename directory `cross/arm-softfp` to `cross/armel`.
Update build scripts, configuration files and CI scripts for arm-softfp
Also fix bug in rootfs of arm-softfp

### CODE REVIEW

**Comment by hseok-oh** [comment]
It does not work only this PR. (will fail ARM build in all PRs)
You need to merge PR #8747, and https://github.com/hseok-oh/coreclr/tree/softfp_to_armel
If you commit additional commit, I will close PR #8747.

**Comment by hqueue** [comment]
@hseok-oh  I've just merged your commits into this PR and let's see CI works well. Thanks!

**Comment by hqueue** [comment]
@dotnet-bot test Linux ARM Emulator Cross Release Build
@dotnet-bot test Linux ARM Emulator Cross Debug Build

**Comment by jyoungyun** [comment]
LGTM

**Comment by hqueue** [comment]
@gkhanna79 PTAL. This replace arm-softfp with armel and we succesfully built CoreCLR for armel at loca host PC.

@mellinoe I saw you merged a PR(https://github.com/dotnet/corefx/pull/14803) in CoreFX. Can you please also take a look at this?

File: tests/scripts/arm32_ci_script.sh [hunk 1]
```diff
@@ -214,8 +214,8 @@ function cross_build_coreclr {
     source "$__ARMEmulRootfs"/dotnet/setenv/setenv_incpath.sh "$__ARMEmulRootfs"
 
     #Apply the changes needed to build for the emulator rootfs
-    (set +x; echo 'Applying cross build patch to suit Linux ARM emulator rootfs')
-    git am < "$__ARMEmulRootfs"/dotnet/setenv/coreclr_cross.patch
+    #(set +x; echo 'Applying cross build patch to suit Linux ARM emulator rootfs')
```
Comment (janvorli) [inline tests/scripts/arm32_ci_script.sh]: If this is no longer needed, could you please delete all the commented out lines instead?

**Comment by gkhanna79** [comment]
LGTM modulo comments from @janvorli. We can merge this once that is addressed.

File: tests/scripts/arm32_ci_script.sh (hunk 1 above)
Comment (hqueue) [inline tests/scripts/arm32_ci_script.sh]: Sure, I will update it.

**Comment by hqueue** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build

**Comment by hqueue** [comment]
@janvorli @gkhanna79 I've update PR with feedback and all CI tests are passed. PTAL.

**Pull request merged**